### PR TITLE
Stable sorts transactions in the block in nonce order, enforce consen…

### PR DESCRIPTION
**Pull request by charithabandi**
…sus limits, induce voteBody and voteID transactions

Support for the following things have been added in this PR:
- Ability to configure and run extensions
- Order Block transactions in nonce order and enforce consensus limits such as maxBlockSz, maxVotesPerTx
- Leader to induce ValidatorVoteBodies Tx if there are any events without resolutions
- Validators can broadcast their vote for the existing resolutions
- ethDeposits extension

### CODE REVIEW

File: extensions/listeners/eth_deposits/deposits.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
+// package ethdeposits implements an listener that listens to Ethereum events
+// and triggers the creation of deposit events in Kwil.
+package ethdeposits
```
Comment (jchappelow) [inline extensions/listeners/eth_deposits/deposits.go:3]: Can we git mv this package instead?

Comment (charithabandi) [reply extensions/listeners/eth_deposits/deposits.go:3]: oh sure!

File: node/txapp/txapp.go [hunk 2]
```diff
@@ -101,18 +100,6 @@ func (r *TxApp) GenesisInit(ctx context.Context, db sql.DB, validators []*types.
...
-	// for _, hook := range hooks.ListGenesisHooks() {
-	// 	err := hook.Hook(ctx, &types.App{
-	// 		Service: r.service.NamedLogger(hook.Name),
-	// 		DB:      db,
-	// 		Engine:  r.Engine,
-	// 	}, chain)
-	// 	if err != nil {
-	// 		return fmt.Errorf("error running genesis hook: %w", err)
-	// 	}
-	// }
```
Comment (jchappelow) [inline node/txapp/txapp.go:114]: oh whoops, conflict here

File: core/types/payloads.go [hunk 3]
```diff
@@ -606,15 +607,67 @@ type ValidatorVoteBodies struct {
 var _ Payload = (*ValidatorVoteBodies)(nil)
 
 func (v *ValidatorVoteBodies) MarshalBinary() ([]byte, error) {
-	return serialize.Encode(v)
+	buf := new(bytes.Buffer)
```
Comment (jchappelow) [inline core/types/payloads.go:610]: ty very much for doing this :pray:

File: node/txapp/txapp.go (hunk 2 above)
Comment (charithabandi) [reply node/txapp/txapp.go]: hahah, rebased.

File: core/rpc/client/admin/jsonrpc/client.go [hunk 4]
```diff
@@ -147,7 +147,6 @@ func (cl *Client) Status(ctx context.Context) (*adminTypes.Status, error) {
 			Syncing:         res.Sync.Syncing,
 		},
 		Validator: &adminTypes.ValidatorInfo{
-			Role:   res.Validator.Role,
```
Comment (jchappelow) [inline core/rpc/client/admin/jsonrpc/client.go:150]: Is this not still helpful to distinguish leader vs regular validator/follower?

File: go.mod [hunk 5]
```diff
@@ -90,7 +108,7 @@ require (
 	github.com/docker/go-units v0.5.0 // indirect
 	github.com/dustin/go-humanize v1.0.1 // indirect
 	github.com/elastic/gosigar v0.14.3 // indirect
-	github.com/ethereum/go-ethereum v1.14.11 // indirect
+	github.com/ethereum/go-ethereum v1.14.11
```
Comment (jchappelow) [inline go.mod:111]: Sorry, very silly nit, but can you move this into the direct imports since it's apparently directly used by deposits (I think) now.

**Review by jchappelow — APPROVED** [review]
LGTM.  Please squash or not as desired and merge when ready.  Thanks for addressing the in-place tx sort issue.

File: core/rpc/client/admin/jsonrpc/client.go (hunk 4 above)
Comment (jchappelow) [reply core/rpc/client/admin/jsonrpc/client.go:150]: Oh, wait, we cannot deduce leader though, can we?

Comment (jchappelow) [reply core/rpc/client/admin/jsonrpc/client.go:150]: Also, what is confusing about it?

**Pull request merged**